ledger/src/blockstore.rs

```diff
@@ -1117,6 +1111,64 @@ impl Blockstore {
             &mut write_batch,
         )?;
 
+        for (erasure_set, working_erasure_meta) in erasure_metas.iter() {
+            if !working_erasure_meta.should_write() {
+                // Not a new erasure meta
+                continue;
+            }
+            let (slot, _) = erasure_set.store_key();
+            // First coding shred from this erasure batch, check the forward merkle root chaining
+            if !self.has_duplicate_shreds_in_slot(slot) {
```

Comment:
Personally rather avoid excessively indented code. This could be:
```rust
if self.has_duplicate_shreds_in_slot(slot) {
   continue;
}
```
to avoid extra indentation.